Dockerfile

```diff
@@ -42,35 +50,48 @@ RUN yarn && yarn cache clean --force
 USER root
 RUN apt-cleanup.sh build-essential
 
-COPY --chown=appuser:appuser docker-entrypoint.sh /entrypoint/docker-entrypoint.sh
-ENTRYPOINT ["/entrypoint/docker-entrypoint.sh"]
+COPY --chown=appuser:appuser . .
+USER appuser
 
 # =============================
-FROM appbase as development
+FROM development_base as development
 # =============================
 
-# Set NODE_ENV to development in the development container
-ARG NODE_ENV=development
-ENV NODE_ENV $NODE_ENV
-
-# Copy in our source code last, as it changes the most
-COPY --chown=appuser:appuser . .
-
 # Bake package.json start command into the image
 CMD ["npm", "start"]
 
+# =============================
+FROM development_base as transpiler
+# =============================
+
+# Transpile Typescript into Javascript
+RUN npm run transpile
+
 # =============================
 FROM appbase as production
 # =============================
 
-# TODO: figure out how to run prod, for now it mirrors dev
+# Copy the transpiled source files to this image
+COPY --from=transpiler --chown=appuser:appuser /app/dist /app/dist
 
-# Set NODE_ENV to production in the production container
+# Set node environment to production to install only main dependencies
 ARG NODE_ENV=production
 ENV NODE_ENV $NODE_ENV
 
-# Copy in our source code last, as it changes the most
-COPY --chown=appuser:appuser . .
+COPY package*.json *yarn* ./
```

Comment:
It looks like it is appuser who is running the yarn command later in this file. Maybe it's better that appuser also owns these files
`COPY --chown=appuser:appuser package*.json *yarn* ./`

Reply:
yes, good idea!